src/linear_algebra/clifford_algebra/spin_group.lean

```diff
@@ -0,0 +1,238 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao
+-/
+import group_theory.group_action.conj_act
+import algebra.star.unitary
+import linear_algebra.clifford_algebra.star
+import linear_algebra.clifford_algebra.even
+
+/-!
+# The Pin group and the Spin group
+
+In this file we define `lipschitz` consisting of all the elements in `(clifford_algebra)ˣ` closed
+under conjugation and construct `pin_group` based on `lipschitz` and `unitary`, and we construct
+`spin_group` based on `pin_group` and `clifford_algebra.even`. Also, we show `pin_group` and
+`spin_group` form a group where the inverse is `star`.
+
+## Main definitions
+
+* `lipschitz`: the Lipschitz group with a quadratic form.
+* `pin_group`: the Pin group with a quadratic form.
+* `spin_group`: the Spin group with a quadratic form.
+
+-/
```

Comment:
I think this should include some discussion of which reference you're using, as well as mentioning the latent ambiguity of definition discussed in https://mathoverflow.net/q/427881/172242 and https://mathoverflow.net/q/251288/172242